components/cloud_start_trial/request_business_email_modal.tsx

```diff
@@ -0,0 +1,158 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useCallback, useEffect, useState} from 'react';
+
+import {FormattedMessage, useIntl} from 'react-intl';
+
+import {trackEvent} from 'actions/telemetry_actions';
+import {validateBusinessEmail} from 'actions/cloud';
+
+import {ItemStatus, TELEMETRY_CATEGORIES} from 'utils/constants';
+
+import GenericModal from 'components/generic_modal';
+import {CustomMessageInputType} from 'components/widgets/inputs/input/input';
+
+import {isEmail} from 'mattermost-redux/utils/helpers';
+
+import StartCloudTrialBtn from './cloud_start_trial_btn';
+import InputBusinessEmail from './input_business_email';
+
+import './request_business_email_modal.scss';
+
+type Props = {
+    onClose?: () => void;
+    onExited: () => void;
+}
+
+const RequestBusinessEmailModal = (
+    {
+        onClose,
+        onExited,
+    }: Props): JSX.Element | null => {
+    const {formatMessage} = useIntl();
+    const [email, setEmail] = useState<string>('');
+    const [customInputLabel, setCustomInputLabel] = useState<CustomMessageInputType>(null);
+    const [trialBtnDisabled, setTrialBtnDisabled] = useState<boolean>(true);
+
+    useEffect(() => {
+        trackEvent(
+            TELEMETRY_CATEGORIES.REQUEST_BUSINESS_EMAIL,
+            'request_business_email',
+        );
+    }, []);
+
+    const handleOnClose = useCallback(() => {
+        if (onClose) {
+            onClose();
+        }
+
+        onExited();
+    }, [onClose, onExited]);
+
+    const handleEmailValues = useCallback((e: React.ChangeEvent<HTMLInputElement>) => {
+        const email = e.target.value;
+        setEmail(email.trim().toLowerCase());
+        validateEmail(email);
+    }, []);
+
+    const validateEmail = useCallback(async (email: string) => {
+        // no value set, no validation and clean the custom input label
+        if (!email) {
+            setCustomInputLabel(null);
+            return;
+        }
+
+        // function isEmail aready handle empty / null value
+        if (!isEmail(email)) {
+            const errMsg = formatMessage({id: 'request_business_email_modal.invalidEmail', defaultMessage: 'This doesnt look like a valid email'});
+            setCustomInputLabel({type: ItemStatus.WARNING, value: errMsg});
+            return;
+        }
+
+        // go and validate the email against the validateBusinessEmail endpoint
+        const isValidBusinessEmail = await validateBusinessEmail()();
+        if (!isValidBusinessEmail) {
+            const errMsg = formatMessage({id: 'request_business_email_modal.not_business_email', defaultMessage: 'This doesnt look like a bussiness email'});
+            setCustomInputLabel({type: ItemStatus.ERROR, value: errMsg});
```

Comment:
Can we add a telemetry event here when the business email check failed? 

@dontoisme any other telemetry you would need for the start trial business email flow in-product ?